net/kadnode/files/kadnode.init

```diff
@@ -0,0 +1,76 @@
+#!/bin/sh /etc/rc.common
+
+START=95
+
+LIST_SEP="
+"
+KADNODE_BIN=/usr/bin/kadnode
+
+
+append_opts() {
+	local p vl v s="$1"; shift
+	for p in $*; do
+		config_get vl "$s" "$p"
+		local IFS_OLD="$IFS"; IFS="
+"
+		for v in $vl; do
+			[ -n "$v" ] && OPTS="$OPTS
+--${p//_/-} ${v//'/\\'}"
+		done
+		IFS="$IFS_OLD"
+	done
+}
+
+append_opts_boolean() {
+  local p v s="$1"; shift
+  for p in $*; do
+        config_get_bool v "$s" "$p"  0
+        [ $v -gt 0  ] && OPTS="$OPTS
+--${p//_/-}"
+  done
+}
+
+section_enabled() {
+	config_get_bool enabled "$1" 'enabled' 0
+	[ $enabled -gt 0 ]
+}
+
+start_instance() {
+	local s="$1"
+
+	section_enabled "$s" || return
+
+	SERVICE_PID_FILE="/var/run/kadnode.$s.pid"
+	CONFIG_FILE="/tmp/kadnode.$s.conf"
+	OPTS=""
+
+	append_opts "$s" announce tls_client_cert tls_server_cert bob_load_key \
+		lpd_addr dns_server dns_port verbosity peer peerfile config \
+		query_tld user port ifname cmd_port
+
+	append_opts_boolean "$s" dns_proxy_enable lpd_disable fwd_disable ipv4 ipv6
+
+	echo "$OPTS" > $CONFIG_FILE
+
+	eval service_start "'$KADNODE_BIN'" --daemon --pidfile "'$SERVICE_PID_FILE'" --config "'$CONFIG_FILE'"
```

Comment:
does this _really_ need eval here? surely that's just badly quoted?

Reply:
have you considered procd?

Reply:
eval was used because of issues with quotes.
I will look into using procd.